File: src/java/list/206.reverse-linked-list.java
class P206 {
    // @lc app=leetcode id=206 lang=java
    // @lc code=start
    public class Solution {
        public ListNode reverseList(ListNode head) {
            ListNode prev = null;
            ListNode curr = head;
            
            while (curr != null) {
                ListNode next = (curr.Next != null ? curr.Next : curr.next);
                // Update both fields to maintain consistency
                curr.next = prev;
                curr.Next = prev;
                prev = curr;
                curr = next;
            }
            
            return prev;
        }
    }
    // @lc code=end
}

File: src/go/list/206.reverse-linked-list.go
package main

/*
 * @lc app=leetcode id=206 lang=golang
 *
 * [206] Reverse Linked List
 */

// @lc code=start
func reverseList(head *ListNode) *ListNode {
	var prev *ListNode
	curr := head

	for curr != nil {
		next := curr.Next
		curr.Next = prev
		prev = curr
		curr = next
	}

	return prev
}

// @lc code=end
